pkg/controller/servicebindingrequest/servicebinder.go

```diff
@@ -246,6 +251,77 @@ func (b *ServiceBinder) onError(
 	return RequeueOnNotFound(err, requeueAfter)
 }
 
+// isApplicationSelectorEmpty returns true if applicationSelector is not declared in
+// the Service Binding Request.
+func isApplicationSelectorEmpty(
+	application v1alpha1.ApplicationSelector,
+) bool {
+	var emptyApplication v1alpha1.ApplicationSelector
+	if application == emptyApplication ||
+		application.ResourceRef == "" &&
+			application.LabelSelector.MatchLabels == nil {
+		return true
+	}
+	return false
+}
+
+// handleApplicationSelectorEmpty handles scenario when applicationSelector is not declared in
+// the Service Binding Request.
+func (b *ServiceBinder) handleApplicationSelectorEmpty() (reconcile.Result, error) {
+	conditionsv1.SetStatusCondition(&b.SBR.Status.Conditions, conditionsv1.Condition{
+		Type:   CollectionReady,
+		Status: corev1.ConditionTrue,
+	})
+	conditionsv1.SetStatusCondition(&b.SBR.Status.Conditions, conditionsv1.Condition{
+		Type:    InjectionReady,
+		Status:  corev1.ConditionFalse,
+		Reason:  EmptyApplicationSelectorReason,
+		Message: ErrEmptyApplicationSelector.Error(),
+	})
+
+	// updating status of request instance
+	sbr, err := b.updateStatusServiceBindingRequest(b.SBR, &b.SBR.Status)
+	if err != nil {
+		return RequeueOnConflict(err)
+	}
+
+	// appending finalizer, should be later removed upon resource deletion
+	sbr.SetFinalizers(append(removeStringSlice(b.SBR.GetFinalizers(), Finalizer), Finalizer))
+	if _, err = b.updateServiceBindingRequest(sbr); err != nil {
+		return NoRequeue(err)
+	}
+
+	b.Logger.Info(ErrEmptyApplicationSelector.Error())
+	return Done()
+}
+
+// handleApplicationSelectorEmpty handles scenario when application is not found.
+func (b *ServiceBinder) handleApplicationNotFound() (reconcile.Result, error) {
+	conditionsv1.SetStatusCondition(&b.SBR.Status.Conditions, conditionsv1.Condition{
+		Type:   CollectionReady,
+		Status: corev1.ConditionTrue,
```

Comment:
I believe the secret should be updated too https://github.com/redhat-developer/service-binding-operator/blob/master/pkg/apis/apps/v1alpha1/servicebindingrequest_types.go#L55 in case of generic binding, or we can have the secret name in Message. If we add in Message then would we need status.secret ?

Reply:
@Avni-Sharma 

[1] Secret is handled here -> https://github.com/redhat-developer/service-binding-operator/pull/498/files#diff-fae660884df4e35af60a04c5ddf930daR335

[2] I don't see a point in adding a secret name in Condition's message when we already have it under status.secret.

Reply:
The status.Secret?